matrix/helper_functions.hpp

```diff
@@ -20,64 +20,39 @@ bool is_finite(Type x) {
 #endif
 }
 
+/**
+ * Wrap value to stay in range [low, high)
+ *
+ * @param x input possibly outside of the range
+ * @param low lower limit of the allowed range
+ * @param high upper limit of the allowed range
+ * @return wrapped value indside the range
+ */
+template<typename Type>
+Type wrap(Type x, Type low, Type high) {
+    Type range = high - low;
+    x = fmod(x - low, range);
```

Comment:
Which fmod is this on NuttX?

In C fmod uses double precision input and out. fmodf uses floats.
In C++ fmod (std::fmod) has multiple overloads for different types (float, double, etc).

Accidentally promoting to double on an stm32f4 is a significant performance penalty that's even worse than the previous code.

Reply:
You're totally right.